File: Examples/VisualBasic/Articles/ManagingWorkbooksWorksheets/ChangeChartDataSource.vb
Imports System.IO
Imports Aspose.Cells

Namespace Articles.ManagingWorkbooksWorksheets
    Public Class ChangeChartDataSource
        Public Shared Sub Run()
            ' ExStart:1
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)

            ' Load sample excel file
            Dim wb As New Workbook(dataDir & Convert.ToString("sample.xlsx"))

            ' Access the first sheet which contains chart
            Dim source As Worksheet = wb.Worksheets(0)

            ' Add another sheet named DestSheet
            Dim destination As Worksheet = wb.Worksheets.Add("DestSheet")

            ' Set CopyOptions.ReferToDestinationSheet to true
            Dim options As New CopyOptions()
            options.ReferToDestinationSheet = True

            ' Copy all the rows of source worksheet to destination worksheet which includes chart as well
            ' The chart data source will now refer to DestSheet
            destination.Cells.CopyRows(source.Cells, 0, 0, source.Cells.MaxDisplayRange.RowCount, options)

            ' Save workbook in xlsx format
            wb.Save(dataDir & Convert.ToString("output_out_.xlsx"), SaveFormat.Xlsx)
            ' ExEnd:1           

        End Sub
    End Class
End Namespace

File: Examples/CSharp/Articles/ManagingWorkbooksWorksheets/ChangeChartDataSource.cs
using System.IO;
using System;
using Aspose.Cells;

namespace Aspose.Cells.Examples.CSharp.Articles.ManagingWorkbooksWorksheets
{
    public class ChangeChartDataSource
    {
        public static void Run()
        {
            //Source directory
            string sourceDir = RunExamples.Get_SourceDirectory();

            //Output directory
            string outputDir = RunExamples.Get_OutputDirectory();

            // Load sample excel file
            Workbook wb = new Workbook(sourceDir + "sampleChangeChartDataSource.xlsx");

            // Access the first sheet which contains chart
            Worksheet source = wb.Worksheets[0];

            // Add another sheet named DestSheet
            Worksheet destination = wb.Worksheets.Add("DestSheet");

            // Set CopyOptions.ReferToDestinationSheet to true
            CopyOptions options = new CopyOptions();
            options.ReferToDestinationSheet = true;

            // Copy all the rows of source worksheet to destination worksheet which includes chart as well
            // The chart data source will now refer to DestSheet
            destination.Cells.CopyRows(source.Cells, 0, 0, source.Cells.MaxDisplayRange.RowCount, options);

            // Save workbook in xlsx format
            wb.Save(outputDir + "outputChangeChartDataSource.xlsx", SaveFormat.Xlsx);

            Console.WriteLine("ChangeChartDataSource executed successfully.");
        }
    }
}
